ryot/src/sprites/error.rs

```diff
@@ -0,0 +1,42 @@
+/*
+ * Ryot - A free and open-source MMORPG server emulator
+ * Copyright (©) 2023 Lucas Grossi <lucas.ggrossi@gmail.com>
+ * Repository: https://github.com/lgrossi/Ryot
+ * License: https://github.com/lgrossi/Ryot/blob/main/LICENSE
+ * Contributors: https://github.com/lgrossi/Ryot/graphs/contributors
+ * Website: https://github.com/lgrossi/Ryot
+ */
+pub type Result<T> = std::result::Result<T, Error>;
```

Comment:
Check out [`color_eyre`](https://docs.rs/color-eyre/latest/color_eyre/) for a general error wrapper that provides a similar, but flexible and colorful `Result`

Reply:
Good point, I added it to my reading list last week and forgot to use it

Reply:
Actually, I did move all of them to use this error, which simplifies a lot. However, I'll leave the usage of color_eyre for a next MR. It's slightly more complex than I thought, when I actually start refactoring the code. Then we can move all errors to reports aligned with color_eyre design.